File: FSM/Entity/InvoiceAddresss/InvoiceAddress.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.InvoiceAddresss.InvoiceAddress
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using System.Collections;
using System.Runtime.CompilerServices;

namespace FSM.Entity.InvoiceAddresss
{
    public class InvoiceAddress
    {
        private DataTypeValidator _dataTypeValidator;
        private bool _exists;
        private bool _deleted;
        private int _InvoiceAddressID;
        private string _Address1;
        private string _Address2;
        private string _Address3;
        private string _Address4;
        private string _Address5;
        private string _Postcode;
        private string _TelephoneNum;
        private string _FaxNum;
        private string _EmailAddress;
        private int _SiteID;

        public InvoiceAddress()
        {
            this._exists = false;
            this._deleted = false;
            this._InvoiceAddressID = 0;
            this._Address1 = string.Empty;
            this._Address2 = string.Empty;
            this._Address3 = string.Empty;
            this._Address4 = string.Empty;
            this._Address5 = string.Empty;
            this._Postcode = string.Empty;
            this._TelephoneNum = string.Empty;
            this._FaxNum = string.Empty;
            this._EmailAddress = string.Empty;
            this._SiteID = 0;
            this._dataTypeValidator = new DataTypeValidator();
        }

        public bool IgnoreExceptionsOnSetMethods
        {
            get
            {
                return this._dataTypeValidator.IgnoreExceptionsOnSetMethods;
            }
            set
            {
                this._dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
            }
        }

        public Hashtable Errors
        {
            get
            {
                return this._dataTypeValidator.Errors;
            }
        }

        public DataTypeValidator DTValidator
        {
            get
            {
                return this._dataTypeValidator;
            }
        }

        public bool Exists
        {
            get
            {
                return this._exists;
            }
            set
            {
                this._exists = value;
            }
        }

        public bool Deleted
        {
            get
            {
                return this._deleted;
            }
        }

        public bool SetDeleted
        {
            set
            {
                this._deleted = value;
            }
        }

        public int InvoiceAddressID
        {
            get
            {
                return this._InvoiceAddressID;
            }
        }

        public object SetInvoiceAddressID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_InvoiceAddressID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Address1
        {
            get
            {
                return this._Address1;
            }
        }

        public object SetAddress1
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Address1", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Address2
        {
            get
            {
                return this._Address2;
            }
        }

        public object SetAddress2
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Address2", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Address3
        {
            get
            {
                return this._Address3;
            }
        }

        public object SetAddress3
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Address3", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Address4
        {
            get
            {
                return this._Address4;
            }
        }

        public object SetAddress4
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Address4", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Address5
        {
            get
            {
                return this._Address5;
            }
        }

        public object SetAddress5
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Address5", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string Postcode
        {
            get
            {
                return this._Postcode;
            }
        }

        public object SetPostcode
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_Postcode", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string TelephoneNum
        {
            get
            {
                return this._TelephoneNum;
            }
        }

        public object SetTelephoneNum
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_TelephoneNum", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string FaxNum
        {
            get
            {
                return this._FaxNum;
            }
        }

        public object SetFaxNum
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_FaxNum", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public string EmailAddress
        {
            get
            {
                return this._EmailAddress;
            }
        }

        public object SetEmailAddress
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_EmailAddress", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int SiteID
        {
            get
            {
                return this._SiteID;
            }
        }

        public object SetSiteID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_SiteID", RuntimeHelpers.GetObjectValue(value));
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/InvoiceAddress/InvoiceAddress.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace InvoiceAddresss

Public Class InvoiceAddress

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "InvoiceAddress Properties"

      
Private _InvoiceAddressID As Integer = 0
Public Readonly Property InvoiceAddressID() As Integer
	Get 
		Return _InvoiceAddressID
	End Get	
End Property	
Public Writeonly Property SetInvoiceAddressID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_InvoiceAddressID", Value)
	End Set
End Property


Private _Address1 As String = String.Empty
Public Readonly Property Address1() As String
	Get 
		Return _Address1
	End Get	
End Property	
Public Writeonly Property SetAddress1() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address1", Value)
	End Set
End Property


Private _Address2 As String = String.Empty
Public Readonly Property Address2() As String
	Get 
		Return _Address2
	End Get	
End Property	
Public Writeonly Property SetAddress2() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address2", Value)
	End Set
End Property


Private _Address3 As String = String.Empty
Public Readonly Property Address3() As String
	Get 
		Return _Address3
	End Get	
End Property	
Public Writeonly Property SetAddress3() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_Address3", Value)
	End Set
End Property

            Private _Address4 As String = String.Empty
            Public ReadOnly Property Address4() As String
                Get
                    Return _Address4
                End Get
            End Property
            Public WriteOnly Property SetAddress4() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Address4", Value)
                End Set
            End Property

            Private _Address5 As String = String.Empty
            Public ReadOnly Property Address5() As String
                Get
                    Return _Address5
                End Get
            End Property
            Public WriteOnly Property SetAddress5() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Address5", Value)
                End Set
            End Property

            Private _Postcode As String = String.Empty
            Public ReadOnly Property Postcode() As String
                Get
                    Return _Postcode
                End Get
            End Property
            Public WriteOnly Property SetPostcode() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Postcode", Value)
                End Set
            End Property


            Private _TelephoneNum As String = String.Empty
            Public ReadOnly Property TelephoneNum() As String
                Get
                    Return _TelephoneNum
                End Get
            End Property
            Public WriteOnly Property SetTelephoneNum() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TelephoneNum", Value)
                End Set
            End Property


            Private _FaxNum As String = String.Empty
            Public ReadOnly Property FaxNum() As String
                Get
                    Return _FaxNum
                End Get
            End Property
            Public WriteOnly Property SetFaxNum() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_FaxNum", Value)
                End Set
            End Property


            Private _EmailAddress As String = String.Empty
            Public ReadOnly Property EmailAddress() As String
                Get
                    Return _EmailAddress
                End Get
            End Property
            Public WriteOnly Property SetEmailAddress() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_EmailAddress", Value)
                End Set
            End Property


            Private _SiteID As Integer = 0
            Public ReadOnly Property SiteID() As Integer
                Get
                    Return _SiteID
                End Get
            End Property
            Public WriteOnly Property SetSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SiteID", Value)
                End Set
            End Property



#End Region

End Class

End Namespace

End Namespace

